README.md

```diff
@@ -1,2 +1,310 @@
-# validator
-A tiny runtime library for type assertions
+# Validator
+
+A tiny library for run time type assertions.
+
+## Installation
+
+```bash
+npm install @mapbox/validator
+```
+
+## Usage
+
+### Basic
+The basic usage 
+
+```javascript
+const v = require('@mapbox/validator');
+
+assertObj = v.assert(
+  v.shape({
+      names: v.arrayOf(v.string)
+  })
+);
+
+assertObj({names: ['ram', 'harry']}); // pass
+assertObj({names: ['john', 987]})// fail
+assertObj({names: 'john'})// fail
+```
+
+### Required
+
+Validation of each value is optional by default, which means passing `null` or `undefined` would not fail the assertion. You should wrap your validator with `v.required` in case you want strict assertion.
```

Comment:
I'm not sure `strict` is the right word here. More specifically, `v.required` means that `null` or `undefined` is not accepted.